Prod reg part 2: implement remanining DSL methods (#17330)
rancher/dashboard: docusaurus/docs/extensions/api/nav/product-registration.md

| `weight` | `number` | No | Side-menu ordering (bigger number on top) |
| `showClusterSwitcher` | `boolean` | No | Show the cluster switcher in navigation |
| `showNamespaceFilter` | `boolean` | No | Show the namespace filter in the header |
| `mapToGroup` | `{ condition: RegExp \| string; group: string }[]` | No | Map resources to specific side-menu groups based on a regex or resource ID. See [mapToGroup](#mapping-resources-to-groups-maptogroup) |
| `ignoreGroups` | `{ regexOrString: string \| RegExp; fn?: (getters: any) => boolean }[]` | No | Hide specific side-menu groups, unconditionally or based on a callback. See [ignoreGroups](#hiding-groups-ignoregroups) |

### `ProductSinglePage`

A product that renders a single full-page Vue component with no side-menu. Extends `ProductMetadata` with:

| Property | Type | Required | Description |
| --- | --- | --- | --- |
| `component` | `RouteComponent` | Yes | Vue component to render for the entire product |

### `ProductChildCustomPage`

> A page inside a product that renders a Vue component you provide. Equivalent to a `virtualType` in the DSL approach.

Reviewer: ```suggestion
A page inside a product that renders a Vue component you provide. Equivalent to a `virtualType` in the legacy DSL approach.
```

(same for other DSL references)

It's worth mentioning DSL for those migrating, but we should make it clear that it's not now an option

Author: updated

| `search` | `string \| boolean` | Path used for search/filtering. Set to `false` to exclude this column from search |
| `formatter` | `string` | Name of a built-in formatter component from `@shell/components/formatter/` or a custom formatter defined in your extension |
| `formatterOpts` | `any` | Configuration options passed to the formatter component |
| `width` | `number` | Fixed column width in pixels. If not set, the column auto-sizes |
| `getValue` | `(row: any) => string \| number \| null` | Custom function to extract or compute the display value from a row. Takes precedence over `value` for display, but `value` is still used for sorting/searching |

### Server-side pagination headers (`sspHeaders`)

When server-side pagination is enabled for a resource type, Rancher Dashboard uses a separate set of headers optimized for paginated data. These headers follow the same format as `headers` but **do not support `getValue`** — since rows are fetched page-by-page from the server, computed client-side values are not available.

> Use `sspHeaders` alongside `headers` to define columns for both pagination modes:

Reviewer: We need info somewhere here that links to https://extensions.rancher.io/extensions/next/performance/scaling/lists#choosing-fields-to-display-sort-and-filter-on, which contains important information on what can be sorted / searched on. Some maybe something like

`Property paths used to display, sort and search on must exist natively on the resource and not in the Dashboard's model for the resource type. For more information please see [here](https://extensions.rancher.io/extensions/next/performance/scaling/lists#choosing-fields-to-display-sort-and-filter-on)`

Author: updated. Thanks for the heads up! 🙏

| `getValue` | `(row: any) => string \| number \| null` | Custom function to extract or compute the display value from a row. Takes precedence over `value` for display, but `value` is still used for sorting/searching |

### Server-side pagination headers (`sspHeaders`)

When server-side pagination is enabled for a resource type, Rancher Dashboard uses a separate set of headers optimized for paginated data. These headers follow the same format as `headers` but **do not support `getValue`** — since rows are fetched page-by-page from the server, computed client-side values are not available.

> **Important:** Property paths used to display, sort and search on must exist natively on the resource and not in the Dashboard's model for the resource type. For more information please see [Choosing fields to display, sort and filter on](https://extensions.rancher.io/extensions/next/performance/scaling/lists#choosing-fields-to-display-sort-and-filter-on).

When `sspHeaders` is defined on a resource page, server-side pagination is automatically enabled for that resource type — no additional configuration is needed.

> > **Important:** Server-side pagination via `sspHeaders` is currently supported for global-level resources (e.g. `provisioning.cattle.io.cluster`). Cluster-scoped resources like `pod` or `service` are not yet supported by `addProduct`. A console warning will be shown if `sspHeaders` is set on a resource type that is not available at the global level.

Reviewer: this isn't quite right. global level resources are resources in the upstream / local cluster, which can contain pods and services. the current implementation uses pods, which are the local cluster pods.

```suggestion
> **Important:** Server-side pagination via `sspHeaders` is currently supported for global-level resources (e.g. `provisioning.cattle.io.cluster`) in the Rancher `local` cluster. Resources in downstream clusters are not yet supported by `addProduct`. A console warning will be shown if `sspHeaders` is set on a resource type that is not available at the global level.
```

| `component` | `RouteComponent` | No | Optional component for the group's own page |
| `weight` | `number` | No | Side-menu ordering (bigger number on top) |

### `ProductChildPage`

A union type representing any page item in a product configuration. It can be one of:

- `ProductChildCustomPage` — a custom page with a Vue component
- `ProductChildResourcePage` — a resource page for a Kubernetes type

```ts
type ProductChildPage = ProductChildCustomPage | ProductChildResourcePage;
```

### `ProductChild`

A union type representing any item that can appear in a product configuration — either a page or a group. This is the type used for arrays passed to `addProduct` and `extendProduct`, and for the `children` property of `ProductChildGroup`.

```ts
type ProductChild = ProductChildGroup | ProductChildPage;
```

## Advanced Configuration

### Table headers (`headers`)

> Resource pages support custom table column headers via the `headers` property. Headers control which columns appear in the resource list view, what data they display, and how they behave (sorting, searching, formatting).

Reviewer: ```suggestion
Resource lists use Headers to control which columns appear in the resource list view, what data they display, and how they behave (sorting, searching, formatting).

By default these mostly come from the resource's definition and it's additionalPrinterColumns field.

When you set `headers` on a resource page, these columns **replace** the default columns.
```

Author: updated
